src/traces/violin/calc.js

```diff
@@ -80,21 +80,27 @@ function silvermanRule(len, ssd, iqr) {
     return 1.059 * a * Math.pow(len, -0.2);
 }
 
-function scottRule(len, ssd) {
-    return ssd * Math.pow(len, -0.2);
-}
-
 function calcBandwidth(trace, cdi, vals) {
-    if(trace.bandwidth) return trace.bandwidth;
-
-    var len = vals.length;
-    var ssd = Lib.stdev(vals, len - 1, cdi.mean);
-    var bw = silvermanRule(len, ssd, cdi.q3 - cdi.q1);
+    var bw;
 
-    if((cdi.max - cdi.min) / bw > 1e5) {
-        bw = scottRule(len, ssd);
+    if(trace.bandwidth) {
+        bw = trace.bandwidth;
+    } else {
+        var len = vals.length;
+        var ssd = Lib.stdev(vals, len - 1, cdi.mean);
+        bw = silvermanRule(len, ssd, cdi.q3 - cdi.q1);
     }
-    return bw;
+
+    // Limit how small the bandwidth can be.
+    //
+    // Silverman's rule of thumb can be "very" small
+    // when IQR does a poor job at describing the spread
+    // of the distribution.
+    // We also want to limit custom bandwidths
+    // to not blow up kde computations.
+    return ((cdi.max - cdi.min) / bw) < 1e5 ?
+        bw :
+        (cdi.max - cdi.min) / 100;
```

Comment:
wait, that's still going to jump from 1/1e5 up to 1/100. I had in mind something like:
```js
if(trace.bandwidth) {
    return Math.max(trace.bandwidth, span / 1e5);
}
else {
    return Math.max(silvermanValue, span / 100);
}
```

Reply:
Oops. My bad -> https://github.com/plotly/plotly.js/pull/2650/commits/c55407079d9006f753156c4446b2bcd4104a1e9a